### PR TITLE
Add connectTimeout to options

### PR DESCRIPTION
**Pull request by ChristianRiesen**
To allow the use of connect timeout option. Default is from curl and thats 300 seconds, which makes near zero sense in any place where this might be used. So having the ability to set this correctly helps the client fail early if need be.

### CODE REVIEW

File: DependencyInjection/FOSElasticaExtension.php [hunk 1]
```diff
@@ -110,6 +110,11 @@ private function loadClients(array $clients, ContainerBuilder $container)
             if (false !== $logger) {
                 $clientDef->addMethodCall('setLogger', array(new Reference($logger)));
             }
+
+            if (isset($clientConfig['connections'][0]['connectTimeout'])) {
+                $clientDef->addMethodCall('setConnectTimeout', [$clientConfig['connections'][0]['connectTimeout']]);
```
Comment (XWB) [inline DependencyInjection/FOSElasticaExtension.php]: This won't work on PHP 5.3, please use `array()` instead.

**Comment by XWB** [comment]
Can you extend the test as well?

**Comment by ChristianRiesen** [comment]
Thanks for the feedback! Will do the changes during the day.

File: DependencyInjection/FOSElasticaExtension.php [hunk 2]
```diff
@@ -110,6 +110,11 @@ private function loadClients(array $clients, ContainerBuilder $container)
             if (false !== $logger) {
                 $clientDef->addMethodCall('setLogger', array(new Reference($logger)));
             }
+
+            if (isset($clientConfig['connections'][0]['connectTimeout'])) {
```
Comment (merk) [inline DependencyInjection/FOSElasticaExtension.php]: Why cant we support all the defined connections?

Comment (ChristianRiesen) [inline DependencyInjection/FOSElasticaExtension.php]: This code was actually not needed. I dug a bit deeper and found that if a config value of connectTimeout is given, that does the trick already and it should set everything correctly inside the underlying elastica library.

**Comment by ChristianRiesen** [comment]
@XWB Found an easier solution and added a test. Please have a look. Thanks!

**Comment by merk** [comment]
Thanks!

**Pull request merged**